### PR TITLE
Add new tabbed dashboard, monitoring with breadcrumb navigation

### PR DESCRIPTION
**Pull request by brian-harris**
* adding new tabbed dashboard to include migrations list
* breadcrumb navigation between migrations list and migration details
* adding SQL DB migration details page
* adding context menu support to migrations list page
* adding common error notification banner and dialog for migration list / migration detail monitoring screens
* bump extension version for release

** New UX ** 
Dashboard tab
image

Migrations list tab
image

Migrations list error detail
image

Migrations list context menu
image

Migration detail with breadcrumb
image

Migration detail with error link and error details dialog
image

SQL DB migration details
image

SQL DB migration table error display
image

### CODE REVIEW

File: extensions/sql-migration/src/api/utils.ts [hunk 1]
```diff
@@ -92,34 +95,67 @@ export function convertTimeDifferenceToDuration(startTime: Date, endTime: Date):
...
+				|| getMigrationStatus(migration)?.toLowerCase().includes(filter)
+				|| getMigrationMode(migration)?.toLowerCase().includes(filter)
+				|| getMigrationTargetType(migration)?.toLowerCase().includes(filter)
+				|| azure.getResourceName(migration.properties.scope)?.toLowerCase().includes(filter)
+				|| azure.getResourceName(migration.id)?.toLowerCase().includes(filter)
+				|| getMigrationDuration(
+					migration.properties.startedOn,
+					migration.properties.endedOn)?.toLowerCase().includes(filter)
+				|| getMigrationTime(migration.properties.startedOn)?.toLowerCase().includes(filter)
+				|| getMigrationTime(migration.properties.endedOn)?.toLowerCase().includes(filter)
```
Comment (AkshayMata) [inline extensions/sql-migration/src/api/utils.ts:136]: def think that filter across columns is needed, just curious about how the performance of this filter looks like. seems like filtering through 10 columns and perf might slow down if we have a lot of rows

File: extensions/sql-migration/src/constants/helper.ts [hunk 2]
```diff
@@ -29,8 +105,8 @@ export function getMigrationTargetType(migration: DatabaseMigration): string {
 	return '';
 }
 
-export function getMigrationTargetTypeEnum(migration: DatabaseMigration): MigrationTargetType | undefined {
-	switch (migration.type) {
+export function getMigrationTargetTypeEnum(migration: DatabaseMigration | undefined): MigrationTargetType | undefined {
```
Comment (AkshayMata) [inline extensions/sql-migration/src/constants/helper.ts:108]: what happens if undefined is passed in?

Comment (AkshayMata) [reply extensions/sql-migration/src/constants/helper.ts:108]: also just curious, why is this needed?

**Review by AkshayMata — APPROVED** [review]

File: extensions/sql-migration/src/constants/helper.ts (hunk 2 above)
Comment (brian-harris) [reply extensions/sql-migration/src/constants/helper.ts:108]: the code that calls this function needs to be able to pass in undefined likely based on the source of the parameter used... the code handles undefined in the default: section.

File: extensions/sql-migration/src/api/utils.ts [hunk 3]
```diff
@@ -92,34 +95,69 @@ export function convertTimeDifferenceToDuration(startTime: Date, endTime: Date):
...
+
+	switch (statusFilter) {
+		case AdsMigrationStatus.ALL:
+			return filteredMigration;
+		case AdsMigrationStatus.ONGOING:
+			return filteredMigration.filter(
+				value => {
+					const status = getMigrationStatus(value);
+					return status === MigrationStatus.InProgress
+						|| status === MigrationStatus.Completing
```
Comment (raymondtruong) [inline extensions/sql-migration/src/api/utils.ts:148]: Do we need 'Completing' here if it already has it already has its own filter type below on line 158?

File: extensions/sql-migration/src/constants/strings.ts [hunk 4]
```diff
@@ -648,20 +666,53 @@ export function STATUS_VALUE(status: string, count: number): string {
...
-	['Completing']: localize('sql.migration.status.completing', 'Completing'),
-	['Canceling']: localize('sql.migration.status.canceling', 'Canceling'),
-	['Failed']: localize('sql.migration.status.failed', 'Failed'),
+	[MigrationStatus.InProgress]: localize('sql.migration.status.inprogress', 'In progress'),
+	[MigrationStatus.Succeeded]: localize('sql.migration.status.succeeded', 'Succeeded'),
+	[MigrationStatus.Creating]: localize('sql.migration.status.creating', 'Creating'),
+	[MigrationStatus.Completing]: localize('sql.migration.status.completing', 'Completing'),
+	[MigrationStatus.Retriable]: localize('sql.migration.status.retriable', 'Retriable'),
+	[MigrationStatus.Canceling]: localize('sql.migration.status.canceling', 'Canceling'),
+	[MigrationStatus.Canceled]: localize('sql.migration.status.canceled', 'Canceled'),
```
Comment (raymondtruong) [inline extensions/sql-migration/src/constants/strings.ts:689]: I notice we use 'Canceled' here, but 'Cancelled' (two Ls) in other places like the legacy status codes - presumably we'll use the spelling with one L everywhere moving forward?

File: extensions/sql-migration/src/constants/strings.ts [hunk 5]
```diff
@@ -761,6 +833,9 @@ export function WARNINGS_COUNT(totalCount: number): string {
 export const AUTHENTICATION_TYPE = localize('sql.migration.authentication.type', "Authentication type");
 
 export const REFRESH_BUTTON_LABEL = localize('sql.migration.status.refresh.label', 'Refresh');
+export const STATUS_LABEL = localize('sql.migration.status.status.label', 'Status');
+export const SORT_LABEL = localize('sql.migration.migration.list.sort.label', 'Sort');
+export const ASSENDING_LABEL = localize('sql.migration.migration.list.assending.label', 'Assending');
```
Comment (raymondtruong) [inline extensions/sql-migration/src/constants/strings.ts:838]: typo, I'm assuming you meant 'Ascending' here

File: extensions/sql-migration/src/dashboard/dashboardTab.ts [hunk 6]
```diff
@@ -0,0 +1,791 @@
...
+			.component();
+
+		linksContainer.addItems(
+			[titleComponent],
+			{ CSSStyles: { 'margin-bottom': '16px' } });
+
+		const links = [
+			{
+				title: localize('sql.migration.dashboard.help.link.migrateUsingADS', 'Migrate databases using Azure Data Studio'),
+				description: localize('sql.migration.dashboard.help.description.migrateUsingADS', 'The Azure SQL Migration extension for Azure Data Studio provides capabilities to assess, get right-sized Azure recommendations and migrate SQL Server databases to Azure.'),
```
Comment (raymondtruong) [inline extensions/sql-migration/src/dashboard/dashboardTab.ts:349]: Can all these strings live in `strings.ts`, or some other resource file-esque store?

Comment (raymondtruong) [reply extensions/sql-migration/src/dashboard/dashboardTab.ts:349]: Moreover, should we turn these links into aka.ms shortlinks, so 1) we can change what they point to without having to change them in the product and also 2) track link hits? The PMs wanted us to do something similar when we introduced these links into DMA so maybe we can check if they want to do something similar here

File: extensions/sql-migration/src/models/stateMachine.ts [hunk 7]
```diff
@@ -34,6 +34,8 @@ export enum State {
 	EXIT,
 }
 
+export const BusinessCriticalTier = 'BusinessCritical';
```
Comment (raymondtruong) [inline extensions/sql-migration/src/models/stateMachine.ts:37]: Should we turn this into an enum and also include GeneralPurpose while we're at it?

File: extensions/sql-migration/src/constants/strings.ts (hunk 4 above)
Comment (brian-harris) [reply extensions/sql-migration/src/constants/strings.ts:689]: yes, I added for legacy references.  It shows up inconsistently at times.

File: extensions/sql-migration/src/constants/strings.ts (hunk 5 above)
Comment (brian-harris) [reply extensions/sql-migration/src/constants/strings.ts:838]: yes, thank you for catching that

File: extensions/sql-migration/src/dashboard/dashboardTab.ts (hunk 6 above)
Comment (brian-harris) [reply extensions/sql-migration/src/dashboard/dashboardTab.ts:349]: I consolidated the strings as suggested, let's create a task to consolidate the URL's as suggested and convert them to fw/akams links

File: extensions/sql-migration/src/models/stateMachine.ts (hunk 7 above)
Comment (brian-harris) [reply extensions/sql-migration/src/models/stateMachine.ts]: good idea

File: extensions/sql-migration/src/api/utils.ts (hunk 3 above)
Comment (brian-harris) [reply extensions/sql-migration/src/api/utils.ts]: good catch

**Review by raymondtruong — APPROVED** [review]

**Pull request merged**